pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} not found in namespace {1!A}.",
+                        classname, namespace))
+
+        clnslist = self._get_subclass_names(classname, class_store, True)
+        clnsdict = NocaseDict()
+        for cln in clnslist:
+            clnsdict[cln] = cln
+
+        clnsdict[classname] = classname
+        return clnsdict
+
+    @staticmethod
+    def _filter_properties(obj, property_list):
+        """
+        Remove properties from an instance or class that aren't in the
+        property_list parameter
+
+        obj(:class:`~pywbem.CIMClass` or :class:`~pywbem.CIMInstance):
+            The class or instance from which properties are to be filtered
+
+        property_list(list of :term:`string`):
+            List of properties which are to be included in the result. If
+            None, remove nothing.  If empty list, remove everything. else
+            remove properties that are not in property_list. Duplicated names
+            are allowed in the list and ignored.
+        """
+        if property_list is not None:
+            property_list = [p.lower() for p in property_list]
+            for pname in obj.properties.keys():
+                if pname.lower() not in property_list:
+                    del obj.properties[pname]
+
+    ###############################################################
+    #
+    #   Access  to data store
+    #
+    #   The following methods provide access to the data store for
+    #   a single namespace. There are separate methods to provide
+    #   access to each of the 3 object stores within each namespace of
+    #   the CIM Repository CIMClass, CIMInstance, and CIMQualifierDeclaration.
+    #
+    ##################################################################
+
+    def get_class_store(self, namespace):
+        """
+        Returns the class object store for the specified CIM namespace within
+        the CIM repository.  This method validates that the namespace exists in
+        the data store.
+
+        This method does not validate that the namespace exists. Be certain
+        namespace is validated against CIM repository before calling this
+        method
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash charactes are ignored.
+
+        Returns:
+
+          Instance of class derived from :class:`~pywbem_mock.`BaseObjectStore`
+          which is the object store for classes in the CIM repository.
+
+        """
+        return self.cimrepository.get_class_store(namespace)
+
+    def get_instance_store(self, namespace):
+        """
+        Returns the instance object store for the specified CIM namespace
+        within the CIM repository.  This method validates that the namespace
+        exists.
```

Comment:
The sentence "This method validates that the namespace exists." is no longer valid and should be removed.